Wip/translation api

Mehrad0711 opened the pull request:
Accommodating https://github.com/stanford-oval/genie-toolkit/pull/806

Mehrad0711 committed:
almond_task: minor fix

Mehrad0711 committed:
server: support custom generation args

gcampax commented on genienlp/server.py:
```diff
@@ -120,8 +121,22 @@ def numericalize_examples(self, ex):
+            )
+            self.numericalizer._tokenizer.src_lang = src_locale
+            self.numericalizer._tokenizer.tgt_lang = tgt_locale
+            self.model.update_language_dependent_configs(tgt_locale)
```
You should avoid accessing _tokenizer from outside the numericalizer class, it's a private property. Add new methods to Numericalizer instead.

gcampax reviewed (approved):
This looks good I think.

Mehrad0711 committed:
predict: assing eval_lang to pred_lang if not provided

Mehrad0711 committed:
example: fix context_plus_question construction if either context or question is empty

Mehrad0711 committed:
almond_translate: accommodate enforcement of unique ids differently

instead change ids on the go to be unique, lifting the burden from user or dataset

Mehrad0711 committed:
server: only allow certain args to be overriden

gcampax commented on setup.py:
```diff
@@ -64,7 +64,7 @@
         'pathos==0.2.8',
         # for kf:
-        'kfserving>=0.5.0',
+        'kfserving==0.5.1',
```
Please don't do this. kfserving depends on ray, but doesn't specify a version bound on it. ray has broken APIs before, breaking kfserving, and us transitively. If we pin kfserving, we remain broken even if kfserving releases a fixed version, until we change genienlp. This creates a cascade effect and takes a lot of work to fix.
(yes python packaging is the worst)

Mehrad0711 replied:
what is the solution then? pinning "ray" instead?

gcampax replied:
There is no good solution, but at least if we leave kfserving unpinned we'll get any fix they make as soon as they make it.

Mehrad0711 replied:
but that will leave our tests broken till then. Apparently ray 1.7.0 was released 3 days ago and had major changes. I suggest pinning ray to 1.6.0 until they patch 1.7.0. what do you think?

Mehrad0711 committed:
setup: pin ray to 1.6.0

Recent release of ray (1.7.0) is breaking our tests due to some import errors.
Will revert this change once a new patch is pushed.
